File: server/models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    studentId: { type: mongoose.Schema.Types.ObjectId, ref: 'Student', required: true },
    razorpayOrderId: { type: String, required: true },
    razorpayPaymentId: { type: String },
    amount: { type: Number, required: true },
    status: { type: String, enum: ['Pending', 'Success', 'Failed'], default: 'Pending' },
    date: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Transaction', transactionSchema);

File: lib/src/models/transaction.dart
class Transaction {
  final String id;
  final double amount;
  final DateTime date;
  final String paymentMethod;
  final String status;
  final String? upiId;

  Transaction({
    required this.id,
    required this.amount,
    required this.date,
    required this.paymentMethod,
    required this.status,
    this.upiId,
  });
}
